### DISCUSSION TITLE
ModuleNotFoundError: No module named 'discord'

### DISCUSSION THREAD

**Discussion by nomxzs** [Rapptz/discord.py#9707] [General]
I copied and pasted the code provided by a YouTuber in Visual Studio Code using Python 3.9.0, but this error code appeared. For reference, I used the same version of Python as the YouTuber. (It was translated using a translator, so there may be grammatical awkwardness.)

**Comment by tailoric** [comment]
just gonna copy paste the common causes for this error and solutions provided from the support discord server's `?tag mnf`

> **Getting `ModuleNotFoundError: No module named 'discord'` but you've installed discord.py?**
Chances are that you've installed to a *different python environment* than you are running your bot from. 

> **Have multiple python versions, such as 3.6, 3.7, and 3.8, installed alongside each other?**
 You probably do. These are different python environments, with their own set of modules/libraries that can be installed by pip. If you used a command starting with `pip install`... you may not even know which python version/environment that *that* `pip` even belongs to.

>Here are some things to try:
▫️  **Be specific:** instead of `pip install`..., do something like `python3.8 -m pip install`...then, run your bot like `python3.8 yourbot.py` - this way, you know you've installed discord.py to python 3.8 and are running your bot with the very same python3.8!
 (on windows this would be more like `py -3.8`...)
▫️  **Run your bot from CLI:** You may love your IDE, but there's no harm in using it like a glorified text editor - you can run the bot in the command prompt/terminal in another window. If you DO want to run from your IDE, make sure to read up on how to configure it with the python you installed discord.py to. 
▫️  **Using pycharm?** It creates per-project virtual environments by default - see `?tag venv` for some information. Not using pycharm, just passing by, or just some random person? You should consider using venvs too!!

Check if any of these help or tell us more information about your setup or your way of running the bot

Also as a side note you should not watch video tutorials for this library they are very often very subpar and/or outdated quickly.
Instead read the [documentation ](https://discordpy.readthedocs.io/en/latest/intro.html) or the [examples](https://github.com/Rapptz/discord.py/tree/master/examples) of this repository
